### ISSUE TITLE
Form.Autosuggest Accessibility - Attributes

### ISSUE THREAD

**Issue by brian-smith-tcril** [openedx/paragon#2438]
### aria-activedescendent

This attribute tells assistive technologies which item in the dropdown is currently focused. The attribute should be attached to the `<input>` element and its value should be the ID of the currently focused item in the list. Currently, we do not require `<Form.AutosuggestOption>` to have an ID, so we would first need to update the component to require an ID or at least create ID's dynamically within `<Form.Autosuggest>`. Then, we'll need to provide functionality in `<Form.Autosuggest>` that utilizes the `onFocus` event for each item in the list. The callback for the `onFocus` event should update `aria-activedescendent`.

We could configure the `onFocus` event similar to how we are already providing an `onClick` event to the child components:
https://github.com/openedx/paragon/blob/d383b57b5db6c1096e38e991d0eb5cbcac5d83a0/src/Form/FormAutosuggest.jsx#L66-L71

### aria-live=”assertive”

This attribute is used to announce something to the user. The “assertive” value tells screen readers that it should take priority over anything else being read. This attribute can be attached to a `<div>` with the text to announce inside. The `<div>` can be inserted above the component as seen in the example. This task involves working with screen readers to ensure the correct number of options are announced to the user. 

```
<ComponentWrapper>
     <div aria-live="assertive" className="sr-only">
	 {dropDownItems.length} options found
     </div>
     {...Component}
<ComponentWrapper>
```
